apis/vshn/v1/dbaas_vshn_keycloak.go

```diff
@@ -0,0 +1,256 @@
+//go:generate yq -i e ../../generated/vshn.appcat.vshn.io_vshnkeycloaks.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.tls.default={})"
+//go:generate yq -i e ../../generated/vshn.appcat.vshn.io_vshnkeycloaks.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.backup.default={})"
+
+// +kubebuilder:object:root=true
+
+// VSHNKeycloak is the API for creating keycloak instances.
+type VSHNKeycloak struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of a VSHNKeycloak.
+	Spec VSHNKeycloakSpec `json:"spec"`
+
+	// Status reflects the observed state of a VSHNKeycloak.
+	Status VSHNKeycloakStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:object:root=true
+type VSHNKeycloakList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []VSHNKeycloak `json:"items,omitempty"`
+}
+
+// VSHNKeycloakSpec defines the desired state of a VSHNKeycloak.
+type VSHNKeycloakSpec struct {
+	// Parameters are the configurable fields of a VSHNKeycloak.
+	Parameters VSHNKeycloakParameters `json:"parameters,omitempty"`
+
+	// WriteConnectionSecretToRef references a secret to which the connection details will be written.
+	WriteConnectionSecretToRef v1.LocalObjectReference `json:"writeConnectionSecretToRef,omitempty"`
+}
+
+// VSHNKeycloakParameters are the configurable fields of a VSHNKeycloak.
+type VSHNKeycloakParameters struct {
+	// Service contains keycloak DBaaS specific properties
+	Service VSHNKeycloakServiceSpec `json:"service,omitempty"`
+
+	// Size contains settings to control the sizing of a service.
+	Size VSHNSizeSpec `json:"size,omitempty"`
+
+	// Scheduling contains settings to control the scheduling of an instance.
+	Scheduling VSHNDBaaSSchedulingSpec `json:"scheduling,omitempty"`
+
+	// TLS contains settings to control tls traffic of a service.
+	TLS VSHNKeycloakTLSSpec `json:"tls,omitempty"`
+
+	// Backup contains settings to control how the instance should get backed up.
+	Backup K8upBackupSpec `json:"backup,omitempty"`
+
+	// Restore contains settings to control the restore of an instance.
+	Restore K8upRestoreSpec `json:"restore,omitempty"`
+
+	// Maintenance contains settings to control the maintenance of an instance.
+	Maintenance VSHNDBaaSMaintenanceScheduleSpec `json:"maintenance,omitempty"`
+}
+
+// VSHNKeycloakServiceSpec contains keycloak DBaaS specific properties
+type VSHNKeycloakServiceSpec struct {
+	// FQDN contains the FQDN which will be used for the ingress.
+	// If it's not set, no ingress will be deployed.
+	// This also enables strict hostname checking for this FQDN.
+	FQDN string `json:"fqdn,omitempty"`
+
+	// RelativePath on which Keycloak will listen.
+	// +kubebuilder:default="/"
+	RelativePath string `json:"relativePath,omitempty"`
```

Comment:
Should be / and not /auth for new installations. `/auth` was legacy for Wildfly.

Reply:
`/` is the default if nothing is specified.

Reply:
Well unfortunately the helm chart keycloakx has `/auth`as default. https://github.com/codecentric/helm-charts/blob/master/charts/keycloakx/values.yaml#L419

Reply:
`// +kubebuilder:default="/"` sets the default when the CR is created.